### ISSUE TITLE
Re-enable header self-containment unit tests with Trilinos

### ISSUE THREAD

**Issue by dalg24** [kokkos/kokkos#3263]
See https://github.com/kokkos/kokkos/pull/3240#discussion_r467341542

**Comment by PhilMiller** [comment]
@ndellingwood Did this ever get done?

**Comment by ndellingwood** [comment]
@PhilMiller no, they are still disabled with Trilinos builds. When I investigated this it wasn't clear to me what the path should be to resolve, here is link to the comment where things left off https://github.com/kokkos/kokkos/pull/3240#discussion_r467341542

**Comment by cwpearson** [comment]
This was fixed in #6164. `add_subdirectory(headers_self_contained)` was guarded by `NOT KOKKOS_HAS_TRILINOS`, and the PR removed that guard.
